### PR TITLE
Homework6

### PR DESCRIPTION
**Pull request by AlexSP0**
with DataPicker

### CODE REVIEW

File: app/src/main/java/com/example/notes/Note.java [hunk 1]
```diff
@@ -1,12 +1,15 @@
 package com.example.notes;
 
+import java.io.Serializable;
 import java.util.Calendar;
 import java.util.GregorianCalendar;
+//Класс заметки
 
-public class Note {
+public class Note implements Serializable {
```
Comment (NIR-Ginko) [inline app/src/main/java/com/example/notes/Note.java:8]: Will you implement the serialization itself?

**Review by NIR-Ginko — CHANGES_REQUESTED** [review]
I suggest to move picture files out of `app/src/main/java/com/example/notes` directory into something like `dist` or `doc`. You can use `git mv` command to do it.

**Comment by NIR-Ginko** [comment]
Nothing to comment at the moment. I suppose I need to try to build and test the code at least once.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -9,6 +9,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.Notes">
+        <activity android:name=".CurrentNoteActivity"></activity>
```
Comment (enotniy) [inline app/src/main/AndroidManifest.xml:12]: <activity android:name=".CurrentNoteActivity" />

File: app/src/main/java/com/example/notes/Note.java [hunk 3]
```diff
@@ -19,9 +22,17 @@ public String getDescription() {
         return description;
     }
 
-    public String getNote() {
+    public String getNoteText() {
         return note;
     }
+
+    public void setDate(Calendar cal) {
+        date=cal.getTimeInMillis();
```
Comment (enotniy) [inline app/src/main/java/com/example/notes/Note.java:30]: форматирование

File: app/src/main/java/com/example/notes/Note.java [hunk 4]
```diff
@@ -19,9 +22,17 @@ public String getDescription() {
...
 
-    public String getNote() {
+    public String getNoteText() {
         return note;
     }
+
+    public void setDate(Calendar cal) {
+        date=cal.getTimeInMillis();
+    }
+    public Calendar getDate() {
```
Comment (enotniy) [inline app/src/main/java/com/example/notes/Note.java:32]: почему у тебя 2 даты в разных форматах? Одна Calendar другая long. Храни обе в Long

File: app/src/main/java/com/example/notes/NotesArray.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package com.example.notes;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+//Класс, который хранит все заметки
+
+public class NotesArray implements Serializable {
+    private ArrayList<Note> notesArray;
```
Comment (enotniy) [inline app/src/main/java/com/example/notes/NotesArray.java:8]: `private List<Note> notesArray;`

File: app/src/main/java/com/example/notes/NotesFragment.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            ShowPortCurrentNote(note);
+        }
+    }
+    private void ShowPortCurrentNote(Note note) {
+        Intent intent = new Intent();
+        intent.setClass(getActivity(), CurrentNoteActivity.class);
+        intent.putExtra(CurrentNoteFragment.NOTE_PARAM, note);
+        startActivity(intent);
+    }
+    private void ShowLandCurrentNote(Note note) {
```
Comment (enotniy) [inline app/src/main/java/com/example/notes/NotesFragment.java:120]: showLandCurrentNote

File: app/src/main/java/com/example/notes/NotesFragment.java [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        }
+    }
+    private void ShowCurrentNote(Note note) {
+        if(isLandscapeFlag) {
+            ShowLandCurrentNote(note);
+        } else {
+            ShowPortCurrentNote(note);
+        }
+    }
+    private void ShowPortCurrentNote(Note note) {
```
Comment (enotniy) [inline app/src/main/java/com/example/notes/NotesFragment.java:114]: showPortCurrentNote

**Review by enotniy — COMMENTED** [review]
Ok for me

**Pull request merged**